pkg/virt-handler/vm.go

```diff
@@ -1,67 +1,72 @@
 package virthandler
 
 import (
-	"fmt"
 	kubeapi "k8s.io/client-go/1.5/pkg/api"
 	kubev1 "k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/rest"
 	"k8s.io/client-go/1.5/tools/cache"
 	"k8s.io/client-go/1.5/tools/record"
 	"kubevirt.io/kubevirt/pkg/api/v1"
 	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/logging"
 	"kubevirt.io/kubevirt/pkg/virt-handler/libvirt"
 )
 
-func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.DomainManager, recorder record.EventRecorder) (cache.Indexer, *cache.Controller) {
+func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.DomainManager, recorder record.EventRecorder, restClient rest.RESTClient) (cache.Indexer, *cache.Controller) {
+	logger := logging.DefaultLogger()
+
 	return kubecli.NewInformer(listWatcher, &v1.VM{}, 0, kubecli.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) error {
-			fmt.Printf("VM ADD\n")
+			logger.Info().Msg("VM ADD")
 			vm := obj.(*v1.VM)
 			err := domainManager.SyncVM(vm)
 			if err != nil {
-				fmt.Println(err)
-				recorder.Event(vm, kubev1.EventTypeWarning, v1.SyncFailed.String(), err.Error())
-				return cache.ErrRequeue{Err: err}
+				goto is_error
 			}
 
-			restClient, err := kubecli.GetRESTClient()
-			if err != nil {
-				return cache.ErrRequeue{Err: err}
+			if vm.Status.Phase != v1.Running {
+				vm.Status.Phase = v1.Running
```

Comment:
You have to copy the object, you might change the cached object otherwise.

Something like model:

```golang
vm := v1.VM{}
// Deep copy the object, so that we can savely manipulate it
model.Copy(&vm, obj)
```